client/src/main/scala/org/apache/celeborn/client/WorkerStatusTracker.scala

```diff
@@ -31,14 +33,26 @@ import org.apache.celeborn.common.protocol.message.StatusCode
 
 class WorkerStatusTracker(
     conf: CelebornConf,
-    lifecycleManager: LifecycleManager,
-    commitManager: CommitManager) extends Logging {
+    lifecycleManager: LifecycleManager) extends Logging {
   private val workerExcludedExpireTimeout = conf.workerExcludedExpireTimeout
+  private val workerStatusListeners = ConcurrentHashMap.newKeySet[WorkerStatusListener]()
 
   // blacklist
   val blacklist = new ShuffleFailedWorkers()
   private val shuttingWorkers: JSet[WorkerInfo] = new JHashSet[WorkerInfo]()
 
+  def registerWorkerStatusListener(workerStatusListener: WorkerStatusListener): Unit = {
+    workerStatusListeners.add(workerStatusListener)
+  }
+
+  def getCandidateBlackListWorkers(): Set[WorkerInfo] = {
+    if (conf.workerExcludedUseAllocatedWorkers) {
```

Comment:
I don't quite understand why return all allocated workers here as blacklist

Reply:
blacklist in master means resource cannot be allocated on, blacklist in lifecycleManager means resource cannot read/write(But this may not complete， Since lifecycleManager do not know a partition data is whether available for reading after commit). As We need to know which worker is not for reading in time, the way for this is compare allocated workers to Master alive workers to get these.  @waitinfuture

Reply:
Have you encountered issues during test if not change this way? The semantics will be changed and the name ```localBlacklist``` will be misleading. I understand the benefit is that LifecycleManager can be notified of unknown workers more timely, but the message body will increase. I'm OK with the change but the name and config description should be refined.